src/Sulu/Bundle/AdminBundle/Resources/js/containers/Application/Application.js

```diff
@@ -23,69 +22,71 @@ type Props = {
-    @observable navigationPinned: boolean = false;
+    @observable navigationState: NavigationState;
 
-    navigationPinnedDisposer: () => void;
+    @computed get navigationPinned() {
+        return this.navigationState === 'pinned';
+    }
 
-    constructor() {
-        super();
+    @computed get navigationVisible() {
+        return this.navigationPinned || this.navigationState === 'visible';
+    }
 
-        intercept(this, 'navigationVisible', (change: IValueWillChange<boolean>) => {
-            if (this.navigationPinned) {
-                log('Cannot change "navigationVisible" while navigation is pinned.');
-                return null;
-            }
+    @action setNavigationState(state: NavigationState) {
+        this.navigationState = state;
+    }
 
-            return change;
-        });
+    set navigationPinned(value: boolean) {
+        this.setNavigationState(value ? 'pinned' : 'hidden');
+    }
 
-        intercept(this, 'navigationPinned', (change: IValueWillChange<boolean>) => {
-            if (change.newValue) {
-                this.navigationVisible = true;
-            }
+    set navigationVisible(value: boolean) {
+        if (this.navigationPinned) {
+            log.warn('Changing the visibility of the navigation is not allowed while navigation is pinned!');
+            return;
+        }
 
-            return change;
-        });
+        this.setNavigationState(value ? 'visible' : 'hidden');
+    }
 
-        observe(this, 'navigationPinned', (change: IValueDidChange<boolean>) => {
-            if (!change.newValue) {
-                this.navigationVisible = false;
-            }
-        });
+    navigationPinnedDisposer: () => void;
+
+    constructor() {
+        super();
 
-        this.setNavigationPinned(!!userStore.getPersistentSetting(NAVIGATION_PINNED_SETTING_KEY));
+        this.navigationPinned = !!userStore.getPersistentSetting(NAVIGATION_PINNED_SETTING_KEY);
 
         this.navigationPinnedDisposer = autorun(
             () => userStore.setPersistentSetting(NAVIGATION_PINNED_SETTING_KEY, this.navigationPinned ? 1 : 0)
         );
     }
 
-    @action toggleNavigation() {
-        this.navigationVisible = !this.navigationVisible;
+    componentWillUnmount() {
+        this.navigationPinnedDisposer();
     }
 
-    @action setNavigationPinned(value: boolean) {
-        this.navigationPinned = value;
+    toggleNavigation() {
+        this.navigationVisible = !this.navigationVisible;
     }
 
     toggleNavigationPinned() {
-        this.setNavigationPinned(!this.navigationPinned);
-    }
-
-    componentWillUnmount() {
-        this.navigationPinnedDisposer();
+        this.navigationPinned = !this.navigationPinned;
     }
 
     handleNavigationButtonClick = () => {
-        if (!this.navigationPinned) {
-            this.toggleNavigation();
+        if (this.navigationPinned) {
+            log.warn('Clicking navigation button is not allowed while navigation is pinned!');
```

Comment:
Is that necessary here? The same warning would be logged when `toggleNavigation` below is called, because it is written a few lines above as well, wouldn't it?

Reply:
This warning tells the developer that he displays the navigation button when it shouldn't be shown. The other warning tells the developer that e.g. some event tries to toggle the navigation. But i can remove it if you want.